From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
Cc: SeongJae Park <sj@kernel.org>, Andrew Morton <akpm@linux-foundation.org>, damon@lists.linux.dev, kernel-team@meta.com, linux-kernel@vger.kernel.org, linux-mm@kvack.org
Subject: [RFC PATCH v2 04/10] mm/damon/paddr: support damos_filter->pass
Message-Id: <20241227210819.63776-5-sj@kernel.org>
In-Reply-To: <20241227210819.63776-1-sj@kernel.org>
Date: Fri, 27 Dec 2024 13:08:13 -0800

'paddr', a DAMON operations set implementation for the physical address
space supports a few types of DAMOS filters (anon, memcg and young), but
ignoring the ->pass field.  Update the logic to support the behavior.

Signed-off-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20241227210819.63776-5-sj@kernel.org
---
 mm/damon/paddr.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index b0c283808ba6..71cb02128b46 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -198,7 +198,7 @@ static unsigned int damon_pa_check_accesses(struct damon_ctx *ctx)
 	return max_nr_accesses;
 }
 
-static bool __damos_pa_filter_out(struct damos_filter *filter,
+static bool damos_pa_filter_match(struct damos_filter *filter,
 		struct folio *folio)
 {
 	bool matched = false;
@@ -237,8 +237,8 @@ static bool damos_pa_filter_out(struct damos *scheme, struct folio *folio)
 	struct damos_filter *filter;
 
 	damos_for_each_filter(filter, scheme) {
-		if (__damos_pa_filter_out(filter, folio))
-			return true;
+		if (damos_pa_filter_match(filter, folio))
+			return !filter->pass;
 	}
 	return false;
 }
-- 
2.39.5